nemo/backends/pytorch/actions.py

```diff
@@ -515,6 +516,18 @@ def _eval(self, tensors_2_evaluate, callback, step, verbose=False):
                 #         self.local_rank, world_size
                 #     )
                 # )
+
+                # Check if we should terminate due to NaN/inf on any workers
+                # We use dtype=int because nccl backend doesn't support torch.bool
+                nan_inf_tensor = torch.tensor(self.nan_or_inf, dtype=int).cuda()
```

Comment:
this will fail on CPU. Can we check if we are on GPU before doing this?

Reply:
Perhaps we should just do a check for distributed. Since this should not be a bug if we are running on a single process, and I do not believe we support distributed CPU processes.

Reply:
This is wrapped within `if dl_nm.placement == DeviceType.AllGpu`; that should take care of this problem, right?
Good catch that it will not stop if there is only a training loop. I did eval to avoid another sync on each step, or a sync on every N steps, but the latter may be a more robust solution. What do you guys think?